air/src/options.rs

```diff
@@ -218,29 +218,42 @@ impl Default for ExecutionOptions {
 }
 
 impl ExecutionOptions {
+    // CONSTANTS
+    // --------------------------------------------------------------------------------------------
+
+    /// The maximum number of VM cycles a transaction is allowed to take.
+    pub const MAX_CYCLES: u32 = 1 << 29;
+
     // CONSTRUCTOR
     // --------------------------------------------------------------------------------------------
 
     /// Creates a new instance of [ExecutionOptions] from the specified parameters.
     ///
-    /// If the `max_cycles` is `None` the maximum number of cycles will be set to `u32::MAX`
+    /// If the `max_cycles` is `None` the maximum number of cycles will be set to 2^29.
     pub fn new(
         max_cycles: Option<u32>,
         expected_cycles: u32,
         enable_tracing: bool,
         enable_debugging: bool,
     ) -> Result<Self, ExecutionOptionsError> {
-        let max_cycles = max_cycles.unwrap_or(u32::MAX);
-        if max_cycles < MIN_TRACE_LEN as u32 {
-            return Err(ExecutionOptionsError::MaxCycleNumTooSmall(expected_cycles));
-        }
-        if max_cycles < expected_cycles {
-            return Err(ExecutionOptionsError::ExpectedCyclesTooBig {
-                max_cycles,
-                expected_cycles,
-            });
-        }
-
+        // Validate max_cycles.
+        let max_cycles = if let Some(max_cycles) = max_cycles {
+            if max_cycles > Self::MAX_CYCLES {
+                return Err(ExecutionOptionsError::MaxCycleNumTooBig {
+                    max_cycles,
+                    max_cycles_limit: Self::MAX_CYCLES,
+                });
+            }
+            if max_cycles < expected_cycles {
+                return Err(ExecutionOptionsError::ExpectedCyclesTooBig {
+                    max_cycles,
+                    expected_cycles,
+                });
+            }
+            max_cycles
+        } else {
+            Self::MAX_CYCLES
+        };
```

Comment:
I'm not sure this is correct: if `max_cycles == None`, won't I be able to set `expected_cycles` to `u32::MAX`?

Reply:
>won't I be able to set expected_cycles to u32::MAX

Are you saying we should be checking expected_cycles to be <= MAX_CYCLES also?

Reply:
Yes, `expected_cycles` should always be less then or equal to `max_cycles` (which could be either user defined or set to `MAX_CYCLES`).